src/main/java/co/cask/gcp/gcs/GCSConfigHelper.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.gcp.gcs;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * A helper class for GCS configs
+ */
+public final class GCSConfigHelper {
+
+  private static final String FILE_SYSTEM_PATH_PREFIX = "gs://";
+
+  private static final Pattern BUCKET_PATTERN = Pattern.compile("gs:\\/\\/(.*?)\\/.*");
+
+  /**
+   * Returns the given path as it is if it starts with {@link GCSConfigHelper#FILE_SYSTEM_PATH_PREFIX} else appends it
+   * in the beginning.
+   *
+   * @param path the path
+   * @return the path with which starts with {@link GCSConfigHelper#FILE_SYSTEM_PATH_PREFIX}
+   */
+  public static String getPath(String path) {
+    return path.startsWith(FILE_SYSTEM_PATH_PREFIX) ? path : FILE_SYSTEM_PATH_PREFIX + path;
```

Comment:
If somebody puts in 'gs:/abc/', this will turn it into 'gs://gs:/abc/'. If somebody puts in '/abc', this will turn into 'gs:///abc'.

I think you can return a URI instead of a String. If it can't be parsed as a URI, an exception will be thrown. If the scheme is not null and not 'gs', throw an exception. If the authority is null (ex: gs:/abc), throw an exception. If the scheme is null, create a new URI with "gs" as the scheme and the schemeSpecificPart from the current URI.

The authority for the URI will then be the bucket and you don't need that pattern matching logic.